modules/kml/src/gpx-loader.ts

```diff
@@ -26,35 +35,46 @@ export const GPXLoader = {
   mimeTypes: ['application/gpx+xml'],
   text: true,
   tests: [GPX_HEADER],
-  parse: async (arrayBuffer, options) =>
+  parse: async (arrayBuffer, options?: GPXLoaderOptions) =>
```

Comment:
We could now type these parse functions with the union of the supported return types from schema module...

Reply:
That won't change anything user-facing, will it? If the user called `GPXLoader.parse` they'd have typing but not if they call `parse(GPXLoader)`, right?

Last Friday I toyed around a bit with trying to add an overload to `load` that would add a generic return type based on the loader definition but couldn't quite get the union return type to work given an array of multiple loaders.

Reply:
Not at this stage, but it would increase the internal rigor.

If you have something that almost works you could always put up a WIP PR.